sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLogSuite.scala

```diff
@@ -209,14 +209,24 @@ class HDFSMetadataLogSuite extends SparkFunSuite with SharedSQLContext {
     }
 
     // Open and delete
-    val f1 = fm.open(path)
-    fm.delete(path)
-    assert(!fm.exists(path))
-    intercept[IOException] {
-      fm.open(path)
+    def closeOnWindows[R <: Closeable](res: => R)(f: => Unit): Unit = {
+      if (Utils.isWindows) {
+        // Windows does not allow deleting opened files.
+        res.close()
```

Comment:
Should we not try to close open file handles in any event, everywhere? It doesn't feel like this must be specific to Windows, even if it only is a problem on Windows.